src/sentry/ingest/consumer_v2/attachment_event.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+
+import logging
+
+import msgpack
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.types import Message
+
+from sentry.ingest.ingest_consumer import (
+    IngestMessage,
+    process_attachment_chunk,
+    process_event,
+    process_individual_attachment,
+    process_userreport,
+)
+from sentry.models import Project
+from sentry.utils import metrics
+
+logger = logging.getLogger(__name__)
+
+
+def decode_and_process_chunks(
+    raw_message: Message[KafkaPayload],
+) -> IngestMessage | None:
+    """
+    The first pass for "attachment" Events:
+
+    - Decode the Kafka payload which is in msgpack format and has a bit of
+      metadata like `type` and `project_id`.
+
+    - Process and save `attachment_chunk`s.
+    - Fetch the `Project` for all other message types.
```

Comment:
Would it make sense to do a bulk fetch for projects (in a fold fashion) in its own step before or after chunks?